SCClassLibrary/Common/Control/ServerStatus.sc

```diff
@@ -28,6 +28,7 @@ ServerStatusWatcher {
 		notified = false;
 		serverBooting = false;
 		this.serverRunning = false;
+		server.changed(\serverRunning);
```

Comment:
I'm pretty sure the problem is here; when this is triggered, the server is considered still running. That's what I see when I set a breakpoint in the C++ code. (https://github.com/supercollider/supercollider/blob/develop/editors/sc-ide/core/sc_server.cpp#L521-L537)

Reply:
this line also causes the same problem as the other call to `server.changed(\serverRunning)`. My solution again is to put it in `{ }.defer` but like I said, that's a hack.

Reply:
ok, got it 
- needed to set hasBooted to false first, since this is what ScIDE now uses. 
  (also synced alive and unresponsive to be false here as well).

> I believe all you have to do is change this line:

```diff
...
@@ -37,7 +37,7 @@ ScIDE {
                serverController = SimpleController(server)
                .put(\serverRunning, { | server, what, extraArg |
                        this.send(\defaultServerRunningChanged, [
-                               server.hasBooted, server.addr.hostname, server.addr.port, server.unresponsive]);
+                               server.serverRunning, server.addr.hostname, server.addr.port, server.unresponsive]);
                })
...
```
OK, done.
I thought showing the server alive as early as possible is good, but I guess it is not worth it.